lib/streamlit/web/server/browser_websocket_handler.py

```diff
@@ -98,12 +98,24 @@ def open(self, *args, **kwargs) -> Optional[Awaitable[None]]:
         except (KeyError, binascii.Error, json.decoder.JSONDecodeError):
             email = "test@example.com"
 
+        userEmailHeader = config.get_option("server.userEmailHeader")
+        if userEmailHeader:
+            email = self.request.headers.get(userEmailHeader)
+
         user_info: Dict[str, Optional[str]] = dict()
         if is_public_cloud_app:
             user_info["email"] = None
         else:
             user_info["email"] = email
 
+        ipheader = config.get_option("server.ipHeader")
+        if ipheader:
+            remote_ip = self.request.headers.get(ipheader)
+        else:
+            remote_ip = self.request.remote_ip
+
+        user_info["ip"] = remote_ip
```

Comment:
The code itself here is fine, but this may need some small tweaks if we maintain the current behavior where this type of information isn't logged unless the developer enables it.